src/erlcloud_aws.erl

```diff
@@ -494,3 +501,189 @@ get_filtered_statuses(ServiceNames, Statuses) ->
                 ServiceNames)
         end,
         Statuses).
+
+
+%%%---------------------------------------------------------------------------
+-spec profile() -> {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
+%% @doc Retrieve a config based on default credentials
+%%
+%% This function will retrieve the credentials for the <em>default</em>
+%% profile, in the same way as the AWS CLI tools, and construct a {@link
+%% #aws_config{}. config record} to be used accessing services.
+%%
+%% @see profile/2
+%%
+profile() ->
+    profile( default, [] ).
+
+
+%%%---------------------------------------------------------------------------
+-spec profile( Name :: atom() ) -> {ok, aws_config()} | {error, string()}.
+%%%---------------------------------------------------------------------------
```

Comment:
According to your implementation, `Name` is either an atom or a binary. It could be better if string also be supported, since binary is not that writing-friendly.